featuretools/entityset/entityset.py

```diff
@@ -1168,124 +1164,127 @@ def add_interesting_values(self, max_values=5, verbose=False, dataframe_name=Non
+                        else:
+                            lti_mask = lti_slice >= time_last - training_window
+                        mask = mask | lti_mask
+                    else:
+                        warnings.warn(
+                            "Using training_window but last_time_index is "
+                            "not set on entity %s" % (dt.id)
+                        )
 
-    #     Args:
-    #         entity_id (str): The id of the entity to query
-    #         instance_vals (pd.Dataframe, pd.Series, list[str] or str) :
-    #             Instance(s) to match.
-    #         variable_id (str) : Variable to query on. If None, query on index.
-    #         columns (list[str]) : Columns to return. Return all columns if None.
-    #         time_last (pd.TimeStamp) : Query data up to and including this
-    #             time. Only applies if entity has a time index.
-    #         training_window (Timedelta, optional):
-    #             Window defining how much time before the cutoff time data
-    #             can be used when calculating features. If None, all data before cutoff time is used.
-    #         include_cutoff_time (bool):
-    #             If True, data at cutoff time are included in calculating features
+                    df = df[mask]
+
+        dt_secondary_time_index = dt.ww.metadata.get('secondary_time_index') or {}
+        for secondary_time_index, columns in dt_secondary_time_index.items():
+            # should we use ignore time last here?
+            df_empty = df.empty if isinstance(df, pd.DataFrame) else False
+            if time_last is not None and not df_empty:
+                mask = df[secondary_time_index] >= time_last
+                if isinstance(df, dd.DataFrame):
+                    for col in columns:
+                        df[col] = df[col].mask(mask, np.nan)
+                elif is_instance(df, ks, 'DataFrame'):
+                    df.loc[mask, columns] = None
+                else:
+                    df.loc[mask, columns] = np.nan
 
-    #     Returns:
-    #         pd.DataFrame : instances that match constraints with ids in order of underlying dataframe
-    #     """
-    #     entity = self[entity_id]
-    #     if not variable_id:
-    #         variable_id = entity.index
+        return df
 
-    #     instance_vals = _vals_to_series(instance_vals, variable_id)
+    def query_by_values(self, dataframe_id, instance_vals, column_id=None, columns=None,
+                        time_last=None, training_window=None, include_cutoff_time=True):
+        """Query instances that have column with given value
 
-    #     training_window = _check_timedelta(training_window)
+        Args:
+            dataframe_id (str): The id of the dataframe to query
+            instance_vals (pd.Dataframe, pd.Series, list[str] or str) :
+                Instance(s) to match.
+            column_id (str) : Column to query on. If None, query on index.
```

Comment:
might be more consistent with the naming convention we've been going with to have the variables be `column_name` and `dataframe_name`

Reply:
Good point - updated.